tutorials/tts/Aligner_Inference_Examples.ipynb

```diff
@@ -0,0 +1,746 @@
+    "        total_frames += durs[token_ind]\n",
+    "        start_frame += durs[token_ind]\n",
+    "\n",
+    "    return dist_sum / total_frames\n",
+    "\n",
+    "# Retrieve the average embedding distances for each pronunciation of \"that\"\n",
+    "that1_mean_dist = get_mean_distance_for_word(\n",
+    "    l2_dists=l2_dists[0],\n",
+    "    durs=disamb_durations[0],\n",
+    "    start_token=1,  # Remember to account for space padding\n",
+    "    num_tokens=len(that1)\n",
+    ")\n",
+    "that2_mean_dist = get_mean_distance_for_word(\n",
+    "    l2_dists=l2_dists[1],\n",
+    "    durs=disamb_durations[1],\n",
+    "    start_token=1,  # Here as well\n",
+    "    num_tokens=len(that2)\n",
+    ")\n",
+    "\n",
+    "print(f\"Average distance for {that1}: {that1_mean_dist}\")\n",
+    "print(f\"Average distance for {that2}: {that2_mean_dist}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6b336cc0",
+   "metadata": {},
+   "source": [
+    "And we're done!\n",
+    "\n",
+    "**With the average distance for `DH AE1 T` being about 377 and the average distance for `DH AH0 T` being about 403, we can pick `DH AE1 T` as the better match.**\n",
+    "\n",
+    "As an exercise, try editing the blocks of code above to disambiguate \"accusation\" (`AE2 K Y AH0 Z EY1 SH AH0 N` versus `AE2 K Y UW0 Z EY1 SH AH0 N`)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c802394c",
+   "metadata": {},
+   "source": [
+    "#### Addendum\n",
+    "\n",
+    "As a side note, there is also a function called `get_mean_dist_by_durations()` that will match up the distances between each token and its corresponding spectrogram frame (using the previously-calculated durations), then calculate the mean over the batch.\n",
+    "\n",
+    "The whole-sentence average may not tell us very much here because we just want to know which pronunciation is closest to what's being said for a specific word, but it's there if you need it as an extra metric!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c138a66f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Mean distance over the batches:\n",
+    "mean_dists = aligner.alignment_encoder.get_mean_dist_by_durations(\n",
+    "    dist=l2_dists.to('cpu'),\n",
+    "    durations=disamb_durations.to('cpu')\n",
+    ")\n",
+    "print(mean_dists)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f8c353c9",
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.9.7"
```

Comment:
any reference link or next steps? Aligner docs, paper? link to the python script to run disambiguation?

Reply:
Good idea, added!